File: main/HardpointType.cpp
#include "HardpointType.h"

enum class HardpointType::HardpointEnum {
    BALLISTIC, ENERGY, MISSILE, AMS, ECM, MASC
};

HardpointType::HardpointType(const HardpointType::HardpointEnum & techBase, const std::string & fullName, const std::string & abbreviation)
    :longName(fullName), shortName(abbreviation), enumId(techBase) {
}

const HardpointType HardpointType::BALLISTIC(HardpointType::HardpointEnum::BALLISTIC, "Ballistic", "B");
const HardpointType HardpointType::ENERGY(HardpointType::HardpointEnum::ENERGY, "Energy", "En");
const HardpointType HardpointType::MISSILE(HardpointType::HardpointEnum::MISSILE, "Missile", "Mi");
const HardpointType HardpointType::AMS(HardpointType::HardpointEnum::AMS, "AMS", "A");
const HardpointType HardpointType::ECM(HardpointType::HardpointEnum::ECM, "ECM", "Ec");
const HardpointType HardpointType::MASC(HardpointType::HardpointEnum::MASC, "MASC", "Ma");

File: mwo-mechbay-entities/src/main/java/com/github/majora_incarnate/mwo/mechbay/entities/enums/HardpointType.java
package com.github.majora_incarnate.mwo.mechbay.entities.enums;

import com.google.gson.annotations.SerializedName;

/**
 * @author Trevin
 */
public enum HardpointType {
    UNKNOWN("Unknown", -1),
    
    @SerializedName("Ballistic")
    BALLISTIC("Ballistic", 0),
    
    @SerializedName("Energy")
    ENERGY("Energy", 1),
    
    @SerializedName("Missile")
    MISSILE("Missile", 2),
    
    @SerializedName("AMS")
    AMS("AMS", 3),
    
    @SerializedName("ECM")
    ECM("ECM", 4),
    
    @SerializedName("MASC")
    MASC("MASC", 5);

    public final String name;
    public final int value;

    HardpointType(final String name, final int value) {
        this.name = name;
        this.value = value;
    }

    @Override
    public String toString() {
        return name;
    }

    public static HardpointType getEnum(String value) {
        switch(value) {
            case "Ballistic":
                return BALLISTIC;
            case "Energy":
                return ENERGY;
            case "Missile":
                return MISSILE;
            case "AMS":
                return AMS;
            case "ECM":
                return ECM;
            case "MASC":
                return MASC;
            default:
                return UNKNOWN;
        }
    }
}
